### PR TITLE
[v7r3] fix: quoting the password for special characters

### LINKED ISSUE #5620: Dirac fails to handle MySQL passwords correctly (at-least at bootstrap)
Hello, 
MySQL 8 enforces password complexity rules.
As a result I chose a MySQL Dirac user  password containing the '@' symbol: `TestPass@X9`
I then ran:
` ./install_site.sh --debug -v v7r3p9 install-v7r3p9.cfg 2>&1 > install-v7r3p9.log
`
There were errors which I grep'd out of the log file:
```
2021-11-18 12:58:25 UTC Framework/ComponentMonitoring ERROR: Could not connect to the database : (_mysql_exceptions.OperationalError) (2005, "Unknown MySQ server host 'X9@localhost' (2)")
2021-11-18 12:58:25 UTC Framework/ComponentMonitoring ERROR: Could not connect to the database : (_mysql_exceptions.OperationalError) (2005, "Unknown MySQ server host 'X9@localhost' (2)")
OperationalError: (_mysql_exceptions.OperationalError) (2005, "Unknown MySQL server host 'X9@localhost' (2)")
2021-11-18 12:58:37 UTC ResourceStatus/ResourceStatus ERROR: Exception while calling initialization function: (_mysql_exceptions.OperationalError) (2005, "Unknown MySQL server host 'X9@localhost' (2)")
OperationalError: (_mysql_exceptions.OperationalError) (2005, "Unknown MySQL server host 'X9@localhost' (2)")
2021-11-18 12:58:37 UTC ResourceStatus/ResourceStatus ERROR: Exception while calling initialization function: (_mysql_exceptions.OperationalError) (2005, "Unknown MySQL server host 'X9@localhost' (2)")

```

So I changed to using '%' instead: `TestPass%X9`
This resulted in errors in the log like this:

```
OperationalError: (_mysql_exceptions.OperationalError) (1045, "Access denied for user 'Dirac'@'localhost' (using password: YES)")
2021-11-18 13:22:55 UTC ResourceStatus/ResourceStatus ERROR: Exception while calling initialization function: (_mysql_exceptions.OperationalError) (1045, "Access denied for user 'Dirac'@'localhost' (using password: YES)")
OperationalError: (_mysql_exceptions.OperationalError) (1045, "Access denied for user 'Dirac'@'localhost' (using password: YES)")
2021-11-18 13:22:55 UTC ResourceStatus/ResourceStatus ERROR: Exception while calling initialization function: (_mysql_exceptions.OperationalError) (1045, "Access denied for user 'Dirac'@'localhost' (using password: YES)")

```

I then changed to using  '-' symbol instead and all proceeded smoothly.

I was less precise about collecting data for the MySQL root user account. I know that '%' works for root but I **think** there may have been problems with '@' but can not recall.

Brief Analysis:
There is some less than robust string handling for MySQL passwords at least in the bootstrap code, possibly more widely.

The '@' symbol is clearly confused by the python code possibly with the '@' symbol used by MySQL to separate user and domain as in the SQL statement:
`CREATE USER 'Dirac'@'localhost' IDENTIFIED BY 'diracpass';
`
Hence the error: `Unknown MySQ server host 'X9@localhost' ` where the X9 is the part of the password string following the '@' sign in the password but is interpreted as the database username... Strange!

The '%' symbol is perhaps more subtle. I wonder if %-formating or other python string formatting is messing with and % in a password string.

Work around:
Avoid '@' in MySQL passwords (Dirac user and root user)
Avoid '%' in Dirac users MySQL password.
The '-' symbol appears to be safe.

I hope that the above is sufficient to recreate the issue and useful to track down and fix the issue.

Regards
Alex Owen

### PR DESCRIPTION
**Pull request by fstagni**
BEGINRELEASENOTES

*Core
FIX: quoting the DB passwords for special characters

ENDRELEASENOTES

closes #5620

### CODE REVIEW

File: src/DIRAC/Core/Utilities/ElasticSearchDB.py [hunk 1]
```diff
@@ -125,6 +126,8 @@ def __init__(
         :param str client_cert: Client certificate.
         """
 
+        password = urlparse.quote_plus(password)
```
Comment (chrisburr) [inline src/DIRAC/Core/Utilities/ElasticSearchDB.py]: Have up you checked if this is correct for open[elastic]search?

**Comment by chrisburr** [comment]
There are more cases: https://github.com/DIRACGrid/DIRAC/blob/1e6e93e4360c6cdd30ca26a29b66b34f73296f72/src/DIRAC/WorkloadManagementSystem/DB/PilotsLoggingDB.py#L72-L73

File: src/DIRAC/Core/Utilities/ElasticSearchDB.py (hunk 1 above)
Comment (fstagni) [reply src/DIRAC/Core/Utilities/ElasticSearchDB.py]: I have read it's correct (did not try to verify).

**Comment by fstagni** [comment]
> There are more cases:
> 
> https://github.com/DIRACGrid/DIRAC/blob/1e6e93e4360c6cdd30ca26a29b66b34f73296f72/src/DIRAC/WorkloadManagementSystem/DB/PilotsLoggingDB.py#L72-L73

This DB should anyway be removed as part of Janusz' work of PilotLogging so I did not bother to update it.

**Review by chrisburr — APPROVED** [review]

**Pull request merged**